app/models/katello/content_view_puppet_environment.rb

```diff
@@ -91,25 +79,16 @@ def archive?
       self.environment.nil?
     end
 
-    def generate_puppet_path(capsule)
-      # rubocop:disable Style/EmptyElse
-      if self.environment
-        File.join(capsule.puppet_path, generate_puppet_env_name, 'modules')
-      else
-        nil #don't generate archived content view puppet environments
-      end
-    end
-
-    def generate_puppet_env_name
-      if self.environment
-        Environment.construct_name(self.organization,
-                                   self.environment,
-                                   self.content_view)
-      end
-    end
-
-    def ignore_global_proxy
-      false
+    def mock_repository_with_root
```

Comment:
We should cache with `@blah ||=` it so we're not just throwing out our new models each access. I'm also not crazy about the name `mock` because it has too much association with tests. I'd prefer something like `tmp_repository_with_root` or `nonpersisted_repository` :man_shrugging: I mostly don't like the `mock` word :)

Reply:
I like nonpersisted,  I'm not entirely sold on the caching those, because these don't even require a db lookup, its super fast

Reply:
Ok, I'm fine with nonpersisted. What about the name?